### ISSUE TITLE
False positive invalid-assignment on tuple[Union] -> Union[Sequence]

### ISSUE THREAD

**Issue by oerc0122** [astral-sh/ty#3426]
### Summary

Slightly odd edge case wherein a single element tuple of a union type is not considered equivalent to the union of the single element tuple types resulting in a false-positive `invalid-assignment`. 

The MWE would be:
```python
from collections.abc import Sequence

def func(c: int | str):
    d: Sequence[int] | Sequence[str] = (c,)
```
However, the playground below is closer to the actual use-case.

https://play.ty.dev/71961565-ec1c-47a6-b43e-3b7592d636e1

I understand the complexity of tuples of union types would be combinatorially scaling, but the known one-element case may be worth expanding.

### Version

0.0.34

**Comment by carljm** [comment]
Thanks for the report! This is #493
